src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -504,3 +504,26 @@ induceEAM p m = es m `AM.overlay` vs m
   where
     es = AM.edges . filter p . AM.edgeList
     vs = AM.vertices . AM.vertexList
+
+-- | Constructs an acyclic graph from any graph based on
+-- a order of vertices to produce an acyclic graph.
+-- The internal order defines the valid set of edges.
+-- 
+-- The order for valid edges is \<, ie. for any two
+-- vertices x and y (x \> y), the only possible edge is (y, x).
+-- This will guarantee the production of an acyclic graph since
+-- no back edges are possible.
+--
+-- For example,
+-- /toAcyclicOrd (1 \* 2 + 2 \* 1) == 1 \* 2/ because
+-- /1 \< 2 == True/ and hence the edge is allowed.
+-- /2 \< 1 == False/ and hence the edge is filtered out.
+--
+-- @
+-- toAcyclicOrd (2 * 1)         == 1 + 2
+-- toAcyclicOrd (1 * 2)         == 1 * 2
+-- toAcyclicOrd (1 * 2 + 2 * 1) == 1 * 2
+-- toAcyclicOrd                 == fromGraph (<) . toGraph
+-- @
```

Comment:
All of this seems very related to the definition of topological ordering, which is, quoting [wikipedia](https://en.wikipedia.org/wiki/Topological_sorting#Relation_to_partial_orders), a "linear extension of a partial order". Maybe saying some words about it is a good idea.

By the way, it seems that enumerating the vertices following the order constraint gives automatically a topological sorting. Having a custom `topsort` for such graphs can be cool, but I don't know if it is simple to implement with the actual module structure.

Reply:
@nobrakal Interesting, isn't `topSort` in this case going to be simply `vertexList`? :)

Reply:
Yes, I think :D

Reply:
Please let me know if the following changes shed more light on topSort.